api/services/sshkeys.go

```diff
@@ -61,6 +79,39 @@ func (s *service) GetPublicKey(ctx context.Context, fingerprint, tenant string)
 }
 
 func (s *service) CreatePublicKey(ctx context.Context, key *models.PublicKey, tenant string) error {
+	exist := func(item string, list []string) bool {
+		for _, elem := range list {
+			if elem == item {
+				return true
+			}
+		}
+
+		return false
+	}
+
+	if err := key.Validate(); err != nil {
+		return ErrPublicKeyInvalid
+	}
+
+	// Checks if the tags in public key are valid.
+	// The tags are valid when they exist in some device.
+	if key.Filter.Tags != nil {
+		if len(key.Filter.Tags) > 0 {
+			tags, _, err := s.store.DeviceGetTags(ctx, tenant)
+			if err != nil {
+				return err
+			}
+
+			for _, tag := range key.Filter.Tags {
+				if !exist(tag, tags) {
```

Comment:
Same here. 

If desired to use a function (which I am fine too) I'd use https://pkg.go.dev/sort#SearchStrings.